pkg/identity/builder.go

```diff
@@ -0,0 +1,71 @@
+package identity
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/identity/store"
+)
+
+const (
+	seedName = "seed.txt"
+)
+
+// NewStore tries to build the best key store available
+// for this ndoe.
+// On a machine with no tpm support, that would be a file
+// store.
+// If TPM is supported, TPM will be used.
+// There is a special case if tpm is supported, but a file seed
+// exits, this file key will be migrated to the TPM store then
+// deleted (only if delete is set to true)
+func NewStore(root string, delete bool) (store.Store, error) {
+	file := store.NewFileStore(filepath.Join(root, seedName))
+	if !store.IsTPMEnabled() {
+		return file, nil
+	}
+
+	// tpm is supported, but do we have a key
+	tpm := store.NewTPM()
+	exists, err := file.Exists()
+	if err != nil {
+		return nil, fmt.Errorf("failed to check for seed file: %s", err)
+	}
+
+	if !exists {
+		return tpm, nil
+	}
+
+	if ok, err := tpm.Exists(); err == nil && ok {
+		// so there is a key on disk, but tpm already has a stored key
+		// then we still just return no need for migration to avoid
+		// overriding the key in tpm
+		return tpm, nil
```

Comment:
What about old data/signatures in this case?

Reply:
This is actually to avoid overriding the key inside tpm if someone decided to put another key on disk. The idea is that key migration will happen one time (and one time only) and then delete the key file. If suddenly a key appeared then it means something is fishy and the new key file is just ignored.

Reply:
Ok, got it. I think we also assume there's no other key at our chosen address, right? (not managed by us).

Reply:
That's actually hard to grantee. Maybe we should add a validation (and clearing step) if the key is invalid.